File: src/main/java/ru/kosdev/signfinder/transform/EqualizeHistTransformer.java
package ru.kosdev.signfinder.transform;

import org.bytedeco.javacpp.opencv_core;
import org.bytedeco.javacpp.opencv_imgproc;
import org.springframework.stereotype.Component;

/**
 * Выравнивание гистограммы
 * http://docs.opencv.org/doc/tutorials/imgproc/histograms/histogram_equalization/histogram_equalization.html
 */
@Component
public class EqualizeHistTransformer implements Transformer {

    @Override
    public void accept(opencv_core.Mat mat) {
        opencv_imgproc.equalizeHist(mat, mat);
    }

}

File: src/main/kotlin/ru/kosdev/signfinder/transform/EqualizeHistTransformer.kt
package ru.kosdev.signfinder.transform

import org.bytedeco.javacpp.opencv_core
import org.bytedeco.javacpp.opencv_imgproc
import org.springframework.stereotype.Component
/**
 * Created by Константин on 27.09.2015.
 */
@Component
public class EqualizeHistTransformer : Transformer {

    override fun accept(mat: opencv_core.Mat) {
        opencv_imgproc.equalizeHist(mat, mat)
    }

}